floodviz/maputils.py

```diff
@@ -0,0 +1,51 @@
+import requests
+
+# function to generate a dictionary of site data after querying NWIS
+def site_dict(site_list, url_prefix):
+    # generate the string of site ids for the url
+    id_input_string = ",".join(site_list)
+
+    # create the url
+    url = url_prefix + "/?format=rdb&sites=" + id_input_string + "&siteStatus=all"
+
+    # get data from url
+    data = ""
+    req = requests.get(url)
+    for line in req.text.splitlines():
+        if not line.startswith("#"):
+            data += line + '\n'
```

Comment:
Consider just making data an array directly rather than reforming the text string and then splitting it into an array. You might be able to use a list comprehension here. Here's a good reference: http://python-3-patterns-idioms-test.readthedocs.io/en/latest/Comprehensions.html